internal/keepers/ocr.go

```diff
@@ -18,17 +18,14 @@ func (k *keepers) Query(_ context.Context, _ types.ReportTimestamp) (types.Query
 // of upkeeps available in and UpkeepService and produces an observation containing upkeeps that
 // need to be executed.
 func (k *keepers) Observation(ctx context.Context, _ types.ReportTimestamp, _ types.Query) (types.Observation, error) {
-	results, err := k.service.SampleUpkeeps(ctx)
+	results, err := k.service.SampleUpkeeps(ctx, k.filter.Filter())
 	if err != nil {
 		return types.Observation{}, err
 	}
 
-	ob := observationMessageProto{
-		RandomValue: k.rSrc.Int63(),
-		Keys:        keyList(filterUpkeeps(results, ktypes.Perform)),
-	}
+	keys := keyList(filterUpkeeps(results, ktypes.Eligible))
```

Comment:
nit. we have two filters (one for in-flight upkeeps and one filter for eligible). Can make the distinction clear in naming